apiary.apib

```diff
@@ -1706,22 +1706,802 @@ root will not be listable, as it is virtual. Access known paths directly.
+        {
+          "Message": "invalid peer address: invalid ipfs address",
+          "Code": 0
+        }
+        ```
+
++ Request With Invalid Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/swarm/disconnect?arg=kitten"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/disconnect?arg=kitten"
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: application/json
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 11 Feb 2016 23:11:48 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "invalid peer address: invalid multiaddr, must begin with /"
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "invalid peer address: invalid multiaddr, must begin with /",
+          "Code": 0
+        }
+        ```
+
++ Request With Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/swarm/disconnect?arg=/ip4/40.78.22.28/tcp/4001/ipfs/QmXsaWN5ygTqHwtfoLk5gX1uvrJrYvgKtwa6TUAh4PpAKW"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/disconnect?arg=/ip4/40.78.22.28/tcp/4001/ipfs/QmXsaWN5ygTqHwtfoLk5gX1uvrJrYvgKtwa6TUAh4PpAKW"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: application/json
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 11 Feb 2016 23:13:35 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (object)
+        - Strings (array)
+            - (string)
+
+    + Body
+
+        ```
+        {
+          "Strings": [
+            "disconnect QmXsaWN5ygTqHwtfoLk5gX1uvrJrYvgKtwa6TUAh4PpAKW success"
+          ]
+        }
+        ```
+
+## filters [GET /swarm/filters]
+List out all currently applied filters. The subcommands can be used
+to add or remove said filters. Filters are specified using the multiaddr-filter format:
+
+**Example:**
+
+```
+/ip4/192.168.0.0/ipcidr/16
+```
+
+Where the above is equivalent to the standard CIDR:
+
+```
+192.168.0.0/16
+```
+
+Filters default to those specified under the `"Swarm.AddrFilters"` config key.
+
++ Request Without Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/swarm/filters"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/filters"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        ```
+
+    + Attributes (object)
+        - Strings (array, nullable)
+            - "/ip4/192.168.0.0/ipcidr/16" (MultiAddr)
```

Comment:
It is not, yet https://github.com/jbenet/multiaddr/blob/master/protocols.csv But it doesn't matter because it can be added in the future, if needed.